### DISCUSSION TITLE
feature: --prefix-paths in `gatsby develop`

### DISCUSSION THREAD

**Discussion by jeetiss** [gatsbyjs/gatsby#16040] [Ideas / Feature Requests]
## Description

seems like --path-prefix in dev mode was broken several versions ago

### Steps to reproduce

- run `gatsby develop --path-prefix`
- await success build
- go to browser

### Expected result

all works

### Actual result

browser tab died after same time
<img width="535" alt="Снимок экрана 2019-07-24 в 12 17 35" src="https://user-images.githubusercontent.com/6726016/61781472-0d275480-ae0d-11e9-8bb6-be680cf63194.png">

### Environment

```
  System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 11.14.0 - /var/folders/yc/_z3g250s2x511_0cvdy1b6s40000gn/T/yarn--1563959887101-0.8550683899292018/node
    Yarn: 1.17.3 - /var/folders/yc/_z3g250s2x511_0cvdy1b6s40000gn/T/yarn--1563959887101-0.8550683899292018/yarn
    npm: 6.9.2 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/local/bin/python
  Browsers:
    Chrome: 75.0.3770.142
    Firefox: 67.0
    Safari: 12.1.1
```

**Comment by DSchau** [comment]
pathPrefix is not currently supported in development mode.

It wouldn't be a _lot_ of work to get it up and running in development, but this is not something that's been a high priority for us. The general philosophy is to use Gatsby components (e.g. `Link` component) which are already pathPrefix aware -- and the built version will be prefixed like you expect.

However -- in an effort of unification and reducing edge-cases/gotchas/etc. I totally agree that this could be valuable work. Would you like to work on making this possible?

**Comment by alexluong** [comment]
Hey @DSchau, can you give a high-level direction on how this should be done?

**Comment by joshuagagne** [comment]
Definitely interested in the high level direction as well. I enabled it quickly in the gatsby-cli's `create-cli.js` for develop mode.  Surprisingly that actually allowed it to add a prefix to the html but none of the assets. The root of the site was `localhost:8000/blogs` but unfortunately `common.js` was still on the `/`.

**Comment by SamIngram96** [comment]
Hello @DSchau @alexluong has there been any progress on this since @joshuagagne 's comment? I would be happy to help develop this feature if everyone else is busy, it's important for our site to work

**Comment by rafael-nogueras** [comment]
@DSchau This is a feature that would be important for me, too. It is problematic to have our development environment setup NOT use the prefix path, while our production builds do, and it seems like it would be a good addition to allow the consistency of the same prefix path also in development. (For what it's worth, it's how the equivalent feature works on Create React App, so it would make that migration path easier for its users).

**Comment by lemonadecreative** [comment]
Can this be reopened? Would love to see this included in the next version.

**Comment by rafael-nogueras** [comment]
Yeah, I second that. We have had SO many bugs related to the prefix path (because it has to be removed when testing in the dev environment, but has to be present in the built environment).

I do not understand the rationale to intentionally having such a significant discrepancy between the development and the built environments -- in fact, it's seems like an anti-pattern to introduce such differences, which can only result in errors that are not detectable during development.

**Comment by LekoArts** [comment]
We won't work on this but would be happy to receive a PR adding this functionality to `gatsby develop`. For contributing you can get started here: https://www.gatsbyjs.org/contributing/

**Comment by jamestharpe** [comment]
I might be able to attempt this next month, once I'm finished with my current project.

**Reply by jhodges10** [reply]
Did you ever get a chance to work on this?

**Reply by jamestharpe** [reply]
Sorry, no - got pulled away onto a non-Gatsby project. I'd love to come back to this, but it's low on my list :(

**Comment by master12** [comment]
bump

**Comment by pietvanzoen** [comment]
bump! :)

**Comment by LekoArts** [comment]
Please don't comment with "bump" or similar terms. GitHub discussions takes the 👍 votes into account for the "Top" sorting / you can share your agreement this way.

**Comment by Taelkir** [comment]
I'm interested in working on this, but not having much luck in finding where to start in the codebase. It doesn't look like gatsby-link needs any changes, and as noted above, the gatsby-cli change seems straightforward. Can any more experienced contributors/maintainers offer some guidance?
